pkg/generators/openapi.go

```diff
@@ -428,29 +428,55 @@ func (g openAPITypeWriter) generate(t *types.Type) error {
 	return nil
 }
 
-func (g openAPITypeWriter) generateExtensions(CommentLines []string) error {
-	extensions := parseExtensions(CommentLines)
-	if len(extensions) == 0 {
-		return nil
-	}
-	// If any extensions exist, then emit code to create them.
-	g.Do("VendorExtensible: spec.VendorExtensible{\nExtensions: spec.Extensions{\n", nil)
-	for _, extension := range extensions {
-		g.Do("\"$.$\": ", extension.name)
-		if extension.hasMultipleValues() {
-			g.Do("[]string{\n", nil)
+func (g openAPITypeWriter) generateStructExtensions(t *types.Type) error {
+	extensions, errors := parseExtensions(t.CommentLines)
+	// Initially, we will only log struct extension errors.
+	if len(errors) > 0 {
+		for e := range errors {
+			glog.V(2).Infof("[%s]: %s\n", t.String(), e)
 		}
-		for _, value := range extension.values {
-			g.Do("\"$.$\",\n", value)
-		}
-		if extension.hasMultipleValues() {
-			g.Do("},\n", nil)
+	}
+	// TODO(seans3): Validate struct extensions here.
+	g.emitExtensions(extensions)
+	return nil
+}
+
+func (g openAPITypeWriter) generateMemberExtensions(m *types.Member, parent *types.Type) error {
+	extensions, errors := parseExtensions(m.CommentLines)
+	// Initially, we will only log member extension errors.
+	if len(errors) > 0 {
+		errorPrefix := fmt.Sprintf("[%s] %s:", parent.String(), m.String())
+		for e := range errors {
+			glog.V(2).Infof("%s %s\n", errorPrefix, e)
 		}
 	}
-	g.Do("},\n},\n", nil)
+	// TODO(seans3): Validate member extensions here.
+	// Example: listType extension is only on a Slice.
+	// Example: cross-extension validation - listMapKey only makes sense with listType=map
+	g.emitExtensions(extensions)
 	return nil
 }
 
+func (g openAPITypeWriter) emitExtensions(extensions []extension) {
+	// If any extensions exist, then emit code to create them.
+	if len(extensions) > 0 {
```

Comment:
Can we early-return instead? `if len(extensions) == 0 { return }` ? or maybe even move this condition to the caller.

Reply:
I will return early. Moving this condition to the caller would be error-prone, since they would need to know not to call it with an empty list.